frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRebaseBranchOntoParentAction.java

```diff
@@ -145,17 +148,21 @@ private void doRebase(AnActionEvent anActionEvent, INonRootManagedBranchSnapshot
     var project = getProject(anActionEvent);
     var gitRepository = getSelectedGitRepository(anActionEvent);
     var gitMacheteRepositorySnapshot = getGitMacheteRepositorySnapshot(anActionEvent);
+    var state = gitRepository.map(r -> r.getState());
+    var isCalledFromContextMenu = anActionEvent.getPlace().equals(ActionPlaces.ACTION_PLACE_CONTEXT_MENU);
+    var shouldExplicitlyCheckout = isCalledFromContextMenu && state.map(s -> Repository.State.DETACHED == s).getOrElse(false);
```

Comment:
Hmm maybe drop a comment what would happen if we don't checkout explicitly while in the detached head state coz I can't see the problem clearly :thinking: